lib/samson/hooks.rb

```diff
@@ -144,8 +144,10 @@ def fire(name, *args)
       end
 
       def render_views(name, view, *args)
-        hooks(name).each_with_object("".html_safe) do |partial, html|
-          html << view.render(partial, *args)
+        hooks(name).each_with_object("".html_safe) do |value, html|
+          partial, params_generator = value
+          new_args = params_generator ? [params_generator.call(*args)] : args
```

Comment:
done.

could not figure out how to test the parameters passed to sub template rendering. so I removed the tests. :|